src/lucky/routable.cr

```diff
@@ -260,7 +260,7 @@ module Lucky::Routable
     end
 
     {% params_with_defaults = PARAM_DECLARATIONS.select do |decl|
-         !decl.value.is_a?(Nop) || decl.type.is_a?(Union) && decl.type.types.last.id == Nil.id
+         !decl.value.is_a?(Nop) || decl.type.is_a?(Union) && decl.type.resolve.nilable?
```

Comment:
Here `resolve.nilable?` *should* be ok :crossed_fingers:  .... but if it starts blowing up apps, then we know where to come back to..